File: 3300-minimum-element-after-replacement-with-digit-sum/3300-minimum-element-after-replacement-with-digit-sum.go
func minElement(nums []int) int {
    ans := make([]int, len(nums))
    for i := 0; i < len(nums); i++ {
        ans[i] = countSumDigit(nums[i])
    }

    minNum := 50
    for _, num := range ans {
        minNum = min(minNum, num)
    }

    return minNum
}

func countSumDigit(n int) int {
    sum := 0
    for n > 0 {
        sum += n % 10
        n /= 10
    }
    return sum
}

// array
// time: O(nd)
// space: O(1)

File: 3300-minimum-element-after-replacement-with-digit-sum/3300-minimum-element-after-replacement-with-digit-sum.py
class Solution:
    def minElement(self, nums: List[int]) -> int:
        a = [0] * len(nums)
        for i in range(len(nums)):
            a[i] = countSumDigit(nums[i])

        ans = 50
        for num in a:
            ans = min(ans, num)

        return ans


def countSumDigit(n):
    sum = 0
    while n > 0:
        sum += n % 10
        n //= 10
    return sum

# array
# time: O(nd)
# space: O(1)